src/csharp/Grpc.Core/Internal/ContextPropagationTokenImpl.cs

```diff
@@ -0,0 +1,119 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Threading;
+
+using Grpc.Core.Utils;
+
+namespace Grpc.Core.Internal
+{
+    /// <summary>
+    /// Implementation of <c>ContextPropagationToken</c> that carries
+    /// all fields needed for context propagation by C-core based implementation of gRPC.
+    /// Instances of <c>ContextPropagationToken</c> that are not of this
+    /// type will be recognized as "foreign" and will be silently ignored
+    /// (treated as if null).
+    /// </summary>
+    internal class ContextPropagationTokenImpl : ContextPropagationToken
```

Comment:
Keeping in line with the scheme in https://github.com/grpc/grpc/pull/17733/files#diff-b6df115f215cae0c5d42d1c311b74233R30, maybe `DefaultContextPropagationToken`?

Reply:
I used `Impl` suffix rather than `Default` to denote that inside Grpc.Core, there is only one specific type accepted as propagation token (that won't get ignored). `Default` sounds to me more like there could be multiple implementations and all of them would be equally acceptable (which is the case for ServerCallContext, but not for propagation token. Hopefully that makes sense.